### PR TITLE
Rework

### PR DESCRIPTION
**Pull request by MarcelBruckner**
- Expert trajectories and pretrained agents now in https://github.com/GAIL-4-BARK/large_data_store
  - They get pulled via bazel with the dependency: data = [ "@com_github_gail_4_bark_large_data_store", ... ]
  - They are added to the gail.py, tfa_generate.py and docs/report BUILDs and the paths to load them are adjusted
- Added examples/tfa_generate.py which we now use to generate sac expert trajectories.
  - Reverted examples/tfa.py to be the original from Patrick
- I removed the command line flags in examples/gail.py, examples/tfa.py and examples/tfa_generate.py. We now set all settings except the mode via the params.json files
  - Removed many launch configs as we now dont use command line arguments anymore
  - Cleaned up and commented the usage of our examples/gail.py and examples/tfa_generate.py scripts
- Load save utils now use relative paths 
  - When training the log dirs are now in the bazel-bin/.../...runfiles/... folders
- Fixed the single_agent_runtime problem with a generic observation space
- If no expert trajectories are loaded (visualize mode) we can now pass None for them and the constructor creates an empty list for compatibility
- TF2RLRunner now creates the necessary model and log dirs
- Added SACRunnerGenerator subclass of SACRunner as suggested for the GenerateExpertTrajectories function
-  Rendered tests are now a py_binary so that they do not fail on bazel test //... as they are anyways no real tests, but to visually confirm the replaying of the dataset
- Removed the pretrained agent from the docs/report as we now use the agent in https://github.com/GAIL-4-BARK/large_data_store
- Some small fixes to the report to be runnable with the new path loading for trajectories and experts

### CODE REVIEW

File: bark_ml/environments/single_agent_runtime.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def step(self, action):
     # step and observe
     self._world.Step(self._step_time)
     if eval_id not in self._world.agents:
-      return [0] * 16, -1, True, None
+      return [0] * self._observer.observation_space.shape[0], -1, True, None
```
Comment (MarcelBruckner) [inline bark_ml/environments/single_agent_runtime.py:65]: This is the generic observation space as Tobias suggested.

File: bark_ml/library_wrappers/lib_tf2rl/runners/gail_runner.py [hunk 2]
```diff
@@ -31,6 +31,12 @@ def __init__(self,
     """
     
     self._expert_trajs = expert_trajs
+    if not self._expert_trajs:
```
Comment (MarcelBruckner) [inline bark_ml/library_wrappers/lib_tf2rl/runners/gail_runner.py:34]: Empty expert trajectories if none are loaded

**Review by marcelbrucker — APPROVED** [review]
I have anything to query. Thanks for taking care of all those issues!

File: bark_ml/tests/py_library_tf2rl_tests/BUILD [hunk 3]
```diff
@@ -64,18 +64,18 @@ py_test(
             "//bark_ml/tests/py_library_tf2rl_tests:base_tests",
         ],
     visibility = ["//visibility:public"],
-    size = "small"
+    size = "small",
```
Comment (pedroacvieira) [inline bark_ml/tests/py_library_tf2rl_tests/BUILD:67]: Should there really be a comma in the end here? The py_binary config bellow doesn't have a comma after the last item

File: examples/tfa.py [hunk 4]
```diff
@@ -37,115 +32,49 @@
...
-                      environment=env,
-                      agent=sac_agent)
-  else:
-    raise ValueError(f'{FLAGS.agent} is no valid agent. See help.')
-
-
+                           render=False)
+
+  # PPO-agent
+  # ppo_agent = BehaviorPPOAgent(environment=env,
```
Comment (pedroacvieira) [inline examples/tfa.py:55]: Why leaving commended code in a delivery code? Is it supposed to be used in the future? If so, why not using a flag or a bool config?

File: examples/tfa.py [hunk 5]
```diff
@@ -37,115 +32,49 @@
...
   if FLAGS.mode == "train":
     runner.SetupSummaryWriter()
     runner.Train()
   elif FLAGS.mode == "visualize":
-    runner.Visualize(FLAGS.num_episodes)
-  elif FLAGS.mode == "generate":
-    expert_trajectories = runner.GenerateExpertTrajectories(num_trajectories=FLAGS.num_trajectories, render=FLAGS.render)
-    save_expert_trajectories(output_dir=output_dir, expert_trajectories=expert_trajectories)
-
+    runner.Visualize(5)
```
Comment (pedroacvieira) [inline examples/tfa.py:73]: Magic number. Isn't better to pass it as a config value / parameter?

File: examples/tfa.py [hunk 6]
```diff
@@ -37,115 +32,49 @@
...
+  
   # store all used params of the training
-  # params.Save(os.path.join(Path.home(), "examples/example_params/tfa_params.json"))
-  sys.exit(0)
+  # params.Save("/home/hart/Dokumente/2020/bark-ml/examples/example_params/tfa_params.json")
```
Comment (pedroacvieira) [inline examples/tfa.py:76]: Do they really want us to set a path to hart's machine hard coded?

**Review by pedroacvieira — COMMENTED** [review]
In the technical side, I didn't see anything wrong :D 
I just commented on some things that seemed a bit odd to me.

File: bark_ml/tests/py_library_tf2rl_tests/BUILD (hunk 3 above)
Comment (MarcelBruckner) [reply bark_ml/tests/py_library_tf2rl_tests/BUILD:67]: The Starlark language doesn't care :+1:

File: examples/tfa.py (hunk 4 above)
Comment (MarcelBruckner) [reply examples/tfa.py:55]: The examples/tfa.py file is just reverted to the state it was created before our fork was made. Tobias requested this. So all code in here is not relevant for us :+1:

**Pull request merged**